src/conn/conn_ckpt.c

```diff
@@ -101,37 +101,23 @@ __ckpt_server(void *arg)
 		if (!__ckpt_server_run_chk(session))
 			break;
 
-		/*
-		 * Checkpoint the database if the connection is marked dirty.
-		 * A connection is marked dirty whenever a btree gets marked
-		 * dirty, which reflects upon a change in the database that
-		 * needs to be checkpointed. Said that, there can be short
-		 * instances when a btree gets marked dirty and the connection
-		 * is yet to be. We might skip a checkpoint in that short
-		 * instance, which is okay because by the next time we get to
-		 * checkpoint, the connection would have been marked dirty and
-		 * hence the checkpoint will not be skipped this time.
-		 */
-		if (conn->modified) {
-			WT_ERR(wt_session->checkpoint(wt_session, NULL));
-
-			/* Reset. */
-			if (conn->ckpt_logsize) {
-				__wt_log_written_reset(session);
-				conn->ckpt_signalled = false;
-
-				/*
-				 * In case we crossed the log limit during the
-				 * checkpoint and the condition variable was
-				 * already signalled, do a tiny wait to clear
-				 * it so we don't do another checkpoint
-				 * immediately.
-				 */
-				__wt_cond_wait(
-				    session, conn->ckpt_cond, 1, NULL);
-			}
-		} else
-			WT_STAT_CONN_INCR(session, txn_checkpoint_skipped);
+		WT_ERR(wt_session->checkpoint(wt_session, NULL));
+
+		/* Reset. */
+		if (conn->ckpt_logsize) {
+			__wt_log_written_reset(session);
```

Comment:
There's a change here in that we'll reset the log-written value even if we decided to skip the checkpoint vs. the previous behavior where we only reset it if `conn.modified` was true. I have no idea if that's interesting or not, but it's a change. @sueloverso?

Reply:
This is an interesting question. If we skip a checkpoint, we should skip reseting how much log was written for the checkpoint. This function is to manage and account for the checkpoint setting of triggering a checkpoint based on log volume rather than time (with the internal thread). If stable isn't moving but log volume grows, this function would continually reset it and checkpoints would not occur at the right time. I think bugs may occur if someone is using timestamps and an internal checkpoint thread or at least we should have a general discussion about that.